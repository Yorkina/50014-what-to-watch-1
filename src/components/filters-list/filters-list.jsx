import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {actionCreator} from '../../reducer';


export const FiltersList = (props) => {
  const {properties, currentFilter, changeFilter} = props;
  const filters = [`All Genre`, ...properties];

  return (
    <ul className="catalog__genres-list">
      {filters.map((filter) => {
        const isActive = filter === currentFilter;

        return (
          <li
            key={filter}
            className={`catalog__genres-item ${isActive && `catalog__genres-item--active`}`}
          >
            <a
              href="#"
              className="catalog__genres-link"
              onClick={(evt) => {
                evt.preventDefault();

                if (!isActive) {
                  changeFilter(filter);
                }
              }}
            >
              {filter}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    properties:
      state.films
        .map((film) => film.genre)
        .filter((genre, ind, arr) => arr.indexOf(genre) === ind),
    currentFilter: state.currentFilter,
  });
};

const mapDispatchToProps = (dispatch) => ({
  changeFilter: (genre) => {
    dispatch(actionCreator.changeGenre(genre));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(FiltersList);

FiltersList.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentFilter: PropTypes.string.isRequired,
  changeFilter: PropTypes.func.isRequired,
};
