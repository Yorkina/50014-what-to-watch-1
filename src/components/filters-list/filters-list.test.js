import React from 'react';
import renderer from 'react-test-renderer';
import {FiltersList} from './filters-list.jsx';


describe(`FiltersList`, () => {
  it(`should render`, () => {
    const props = {
      properties: [`Comedies`, `Dramas`, `Crime`],
      currentFilter: `Comedies`,
      changeFilter: jest.fn(),
    };

    const tree = renderer.create(<FiltersList {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
