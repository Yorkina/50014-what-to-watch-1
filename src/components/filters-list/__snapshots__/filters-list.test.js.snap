// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersList should render 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All Genre
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
</ul>
`;
